turbo/snapshotsync/freezeblocks/block_snapshots.go

```diff
@@ -2504,7 +2511,13 @@ func (s *RoSnapshots) ViewSingleFile(t snaptype.Type, blockNum uint64) (segment
 		return nil, false, noop
 	}
 
-	segmentRotx := segs.BeginRotx()
+	segmentRotx := func() *segmentsRotx {
+		// BeginRo increments refcount - which is contended
```

Comment:
refcount - is atomic. and `dirtySegmentsLock` guarding only `dirtySegments` field